### DISCUSSION TITLE
Revisiting BALSAMIC QC thresholds

### DISCUSSION THREAD

**Discussion by mathiasbio** [Clinical-Genomics/BALSAMIC#1259] [General]
Currently there are often cases failing a particular QC metric in BALSAMIC and which is soon after set as PASS. It seems it is time to revisit the thresholds we have set, and see if they ought to be changed, or replaced, or if we can give feedback to the lab for adjustments, etc! 

I'm making this discussion to save the progress of the investigation into the QC-land, and to leave the floor open for feedback and discussions. 

Current QC thresholds:

| WGS                          |                                                               |
|------------------------------|---------------------------------------------------------------|
| PCT_PF_READS_IMPROPER_PAIRS: | <0.1                                                          |
| PCT_15X:                     | >0.95 (normal)                                                |
| PCT_60X:                     | >0.8 (tumor)                                                  |
| NUMBER_OF_SITES SNVs:        | <50000                                                        |
| NUMBER_OF_SITES SVs:         | <50000                                                        |
| RELATEDNESS:                 | >0.8                                                          |
|                              |                                                               |
| TGA                          |                                                               |
| PCT_TARGET_BASES_20X:        | >0.95 (twistexome, twistexomerefseq, twistexomecomprehensive) |
| PCT_TARGET_BASES_250X:       | >0.95 (gmcksolid, gmssolid, probio)                           |
| PCT_TARGET_BASES_500X:       | >0.95 (gmslymphoid, gmsmyeloid), >0.9 (lymphoma, lymphomatic) |
| PCT_TARGET_BASES_1000X:      | >0.95 (gicfdna, hdcfdna, swep53cfDNA)                         |
| GC_DROPOUT:                  | <1                                                            |
| RELATEDNESS:                 | >0.8                                                          |
| NUMBER_OF_SITES SNVs:        | <50000                                                        |
| NUMBER_OF_SITES SVs:         | <50000                                                        |

A sheet with collections of common QC fails in balsamic (probably not complete)
[Balsamic QC fails](https://docs.google.com/spreadsheets/d/1dgkrKYKFYBAF8Mmq8OtBlFQ2b6a8_WvtKDhQyZZxxn4/edit#gid=0)

In summary one can say that we're mostly looking at horizontal coverage as a QC-threshold, which makes sense as this value is directly related to our ability to detect variants in the regions of interest, at the desired minimum VAF. 

There are a few other QC-metrics that are not strictly related to coverage:
- PCT_PF_READS_IMPROPER_PAIRS: Which has been useful in detecting problematic samples which fail SV analysis due to too many chimeric and improper paired reads. 
- RELATEDNESS: From somalier to capture sample mixups.
- GC_DROPOUT: Detection of loss of regions with more extreme levels of %GC-content
- NUMBER_OF_SITES SNVs & SVs: ...

At the moment it's hard to set clear goals from this project of looking at the QC-thresholds. 

**The coverage metrics are difficult to adjust!** 

They are set where they are because the clinicians that order the analysis want to be able to find VAFs at a particular frequency. So we can't just lower them...

But we can still learn some things from looking at the QC metrics:
- We may learn that frequently there are samples with lots of CNVs, which affect the evenness of coverage in an unpredictable way before analysis. And we may reduce the number of cases that fail coverage-thresholds simply by sequencing more, even if cases with normal levels of evenness already pass the thresholds.
- We may discover that the coverage is really co-varying with a particular lab metrics and try to make some recommendations to the lab and in that way reduce the number of samples that fail the coverage.
- ...

**Other QC metrics**

Comment on expectations for changing current QC thresholds:
- PCT_PF_READS_IMPROPER_PAIRS: I think has been pretty spot on, and has been used to detect problematic samples from a particular customer which ends up having issues with SV variant calling. I don't think we will end up adjusting this.
- RELATEDNESS: Will probably also remain the same!
- NUMBER_OF_SITES SNVs & SVs: I don't think this is the type of QC metric we would change the threshold for. 
-  GC_DROPOUT: This seems to be a metric that fails a lot! And which we may learn a lot about by looking into the distributions of this value across multiple samples. Maybe we can give feedback to the lab, maybe we learn that this value is generally always high for a particular type of analysis and we can optimise the threshold to be tailored for different analyses. 

Maybe we realize that there is some QC metric that ought to be implemented that hasn't been.

### Investigate QC metrics across samples

First it seems that someone just needs to start collecting the stats from the multiqc_data.json saved in housekeeper and try to get a better picture of:
- Which QC stats fail the most often
- What's the distribution like for these metrics
- Are there any linked QC stats from the lab?

**Comment by pbiology** [comment]
So how should we keep working on this matter? Do we need to track it somewhere else?

**Comment by mathiasbio** [comment]
Made issue here: https://github.com/Clinical-Genomics/BALSAMIC/issues/1297
